utils.js

```diff
@@ -0,0 +1,59 @@
+function mungeLocation(locationData) {
+    try {
+        const firstItem = locationData[0];
+
+        return {
+            formatted_query: firstItem.display_name,
+            latitude: firstItem.lat,
+            longitude: firstItem.lon
+        };
+    } catch (e) {
+        return {};
+    }
+}
+function mungeWeather(weatherData) {
+    try {
+        const transformedData = weatherData.data.map((forecast) => {
+            
+            return {
+                forecast: forecast.weather.description,
+                time: forecast.valid_date
+            };
+        });
+
+        return transformedData.slice(0, 8);
+    } catch (e) {
+        return [{}];
+    }
+}
+function mungeTrails(trailData) {
+    try {
+        const transformedData = trailData.trails.map((hike) => {
+            
+            return {
```

Comment:
You've got some naming issues here that are most likely the cause of your undefined problems.
starVotes --> star_votes
url --> trail_url
conditionDetails --> condition_details
conditionDate --> condition_date

Reply:
You are mapping the data returned from the api here, so you have to use their naming conventions on the left side of the colon...you can use your own on the right.